### DISCUSSION TITLE
What are the benefits of passing a proxy using context?

### DISCUSSION THREAD

**Discussion by leochiu-a** [pmndrs/valtio#788] [Q&A] [answered]
I saw in the document that there are two ways to pass proxy-state, one is using modules and the other is using context. Using modules seems to be better for maintaining actions and states, so does passing proxy state using context have any clear advantages?

**Comment by dai-shi** [comment, marked as answer]
You can reuse a component. Here's an example with props. (Context usage is basically the same.)

```jsx
const Component = ({ childState }) => {
  const snap = useSnapshot(childState);
  // ...
};

const Parent = () => {
  return (
    <>
      <Component childState={state.foo} />
      <Component childState={state.bar} />
    </>
  );
};
```
